src/main/scala/com/raphtory/core/config/PulsarController.scala

```diff
@@ -116,9 +140,9 @@ class PulsarController(conf: Config) {
       .topic(topic)
       .enableBatching(true)
       .batchingMaxPublishDelay(1, TimeUnit.MILLISECONDS)
-      .batchingMaxMessages(Integer.MAX_VALUE)
+      .batchingMaxMessages(10000)
```

Comment:
I believe this was set based on pulsar best practices.
The max Pending Messages I believe can just be set back to default though? - @Alnaimi- can you confirm.

Reply:
The `maxPendingMessages` change is because @Haaroon actually saw a deadlock after I changed the flushing of vertex messages to make sure they are actually sent before sending the `GraphFunctionComplete` message, i.e., the flushAsync call would never complete. With this set to default value, the old behaviour of looping in CheckMessages forever came back before I set the subscription to exclusive.